contentcuration/contentcuration/assets/shared/views/PrimaryDialog.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <VDialog
+    v-model="open"
+    :disabled="disabled"
+    attach="body"
+    maxWidth="290"
+  >
+    <VCard>
+      <VCardTitle
+        v-if="title"
+      >
+        {{ title }}
+      </VCardTitle>
+      <VCardText>
+        <slot></slot>
+      </VCardText>
+      <VCardActions>
+        <slot name="actions"></slot>
+      </VCardActions>
+    </VCard>
+  </VDialog>
+</template>
+
+<script>
+
+  import Vue from 'vue';
+
+  const modalTracker = Vue.observable({
```

Comment:
This tracks if a primary modal is currently open. If so, code below prevents the modal from opening.

Reply:
The `modalTracker` lives in the module itself, so it lives on as long as the JS scope that it is executed in goes on. I think this would probably work even with multiple Vue root instances, but this is designed for the `one Vue to rule them all` scenario.